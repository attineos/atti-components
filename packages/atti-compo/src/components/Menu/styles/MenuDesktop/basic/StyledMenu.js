import styled, { css } from 'styled-components'

const colors = css`
  background: ${({ theme }) => theme.components.menu.menuDesktop.basic.menu.colors.background};
`

const sizes = css`
  height: ${({ theme }) => theme.components.menu.menuDesktop.withButtons.menu.sizes.height};
  width: ${({ theme }) => theme.components.menu.menuDesktop.withButtons.menu.sizes.width};
`

const StyledMenu = styled.nav`
  ${colors};
  ${sizes};

  display: flex;
  align-items: center;
`

export default StyledMenu
