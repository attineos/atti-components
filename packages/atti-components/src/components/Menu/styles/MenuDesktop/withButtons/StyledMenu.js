import styled, { css } from 'styled-components'

const colors = css``

const sizes = css`
  height: ${({ theme }) => theme.components.menu.menuDesktop.withButtons.menu.sizes.height};
  width: ${({ theme }) => theme.components.menu.menuDesktop.withButtons.menu.sizes.width};
`

const StyledMenu = styled.nav`
  ${colors};
  ${sizes};

  display: flex;
  align-items: center;

  li {
    flex-grow: 1;
    border-right: 1px solid
      ${({ theme }) => theme.components.menu.menuDesktop.withButtons.menu.colors.border};
  }
  li:last-child {
    border-right: none;
  }
`

export default StyledMenu
