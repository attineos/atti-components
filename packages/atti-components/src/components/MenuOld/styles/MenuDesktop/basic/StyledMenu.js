import styled, { css } from 'styled-components'

const colors = css`
  background: ${({ theme }) => theme.components.menuOld.menuDesktop.basic.menu.colors.background};
`

const sizes = css`
  height: ${({ theme }) => theme.components.menuOld.menuDesktop.withButtons.menu.sizes.height};
  width: ${({ theme }) => theme.components.menuOld.menuDesktop.withButtons.menu.sizes.width};
`

const StyledMenu = styled.nav`
  ${colors};
  ${sizes};

  display: flex;
  align-items: center;
`

export default StyledMenu
