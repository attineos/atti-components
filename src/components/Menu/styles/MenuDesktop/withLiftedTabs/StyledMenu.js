import styled, { css } from 'styled-components'

const colors = css``

const sizes = css`
  height: ${({ theme }) => theme.components.menu.menuDesktop.withLiftedTabs.menu.sizes.height};
  width: ${({ theme }) => theme.components.menu.menuDesktop.withLiftedTabs.menu.sizes.width};
`

const StyledMenu = styled.nav`
  ${colors};
  ${sizes};

  display: flex;
  align-items: center;

  li {
    flex-grow: 1;
  }
  li:first-child {
    border-left: none;
  }
  li:last-child {
    border-right: none;
  }
`

export default StyledMenu
