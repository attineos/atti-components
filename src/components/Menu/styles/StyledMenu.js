import styled, { css } from 'styled-components'

const colors = css`
  background: ${({ theme }) => theme.components.menu.menu.colors.background};
`

const sizes = css`
  height: ${({ theme }) => theme.components.menu.menu.sizes.minHeight};
`

const StyledMenu = styled.nav`
  ${colors};
  ${sizes};

  display: flex;
  align-items: center;
`

export default StyledMenu
