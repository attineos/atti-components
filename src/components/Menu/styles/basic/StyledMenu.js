import styled, { css } from 'styled-components'

const colors = css`
  background: ${({ theme }) => theme.components.menu.basic.menu.colors.background};
`

const sizes = css`
  height: ${({ theme }) => theme.components.menu.withButtons.menu.sizes.height};
  width: ${({ theme }) => theme.components.menu.withButtons.menu.sizes.width};
`

const StyledMenu = styled.nav`
  ${colors};
  ${sizes};

  display: flex;
  align-items: center;
`

export default StyledMenu
