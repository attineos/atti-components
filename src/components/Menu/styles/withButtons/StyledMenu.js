import styled, { css } from 'styled-components'

const colors = css``

const sizes = css`
  height: ${({ theme }) => theme.components.menu.withButtons.menu.sizes.height};
  width: ${({ theme }) => theme.components.menu.withButtons.menu.sizes.width};
`

const StyledMenu = styled.nav`
  ${colors};
  ${sizes};

  display: flex;
  align-items: center;

  border: 1px solid ${({ theme }) => theme.components.menu.withButtons.menu.colors.border};
  border-radius: ${({ theme }) => theme.components.menu.withButtons.menu.borderRadius};

  li {
    flex-grow: 1;
    border-right: 1px solid ${({ theme }) => theme.components.menu.withButtons.menu.colors.border};
  }
  li:last-child {
    border-right: none;
  }
`

export default StyledMenu
