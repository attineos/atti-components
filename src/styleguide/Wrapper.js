import React from 'react'
import PropTypes from 'prop-types'
import { remove, get } from 'lodash'

import Cookies from './helpers/Cookies'
import themes from './config/themes'

import ThemeProvider from '../components/ThemeProvider'

const updateWrappers = []

export const getWrappersToUpdate = () => updateWrappers

class Wrapper extends React.Component {
  static propTypes = {
    /**
     * The child node.
     */
    children: PropTypes.node.isRequired,
  }

  state = {
    selectedTheme: null,
  }

  componentWillMount() {
    updateWrappers.push(this)

    const selectedTheme = Cookies.hasItem('selectedTheme')
      ? Cookies.getItem('selectedTheme')
      : 'default'

    this.state.selectedTheme = get(themes, selectedTheme, null)
  }

  componentWillUnmount() {
    remove(updateWrappers, item => item === this)
  }

  handleSetTheme = newTheme =>
    this.setState(
      {
        selectedTheme: newTheme,
      },
      () => this.forceUpdate(),
    )

  render() {
    const { children } = this.props
    const { selectedTheme } = this.state

    return (
      <ThemeProvider theme={selectedTheme} key={Math.random()}>
        {children}
      </ThemeProvider>
    )
  }
}

export default Wrapper
