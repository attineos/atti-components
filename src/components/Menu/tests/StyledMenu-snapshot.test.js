/* This file was generated automatically, please do not write in it. */
import React from 'react';
import StyledMenu from '../styles/StyledMenu';
import Wrapper from 'styleguide/Wrapper'
import renderer from 'react-test-renderer';

const mock = () => {}

it('StyledMenu renders correctly', () => {
  const tree = renderer
    .create(
      <Wrapper>
        <StyledMenu ></StyledMenu>
      </Wrapper>
      )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
    