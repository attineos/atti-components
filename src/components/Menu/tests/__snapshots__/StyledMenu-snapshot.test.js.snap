// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StyledMenu renders correctly 1`] = `
<nav
  className="sc-bdVaJa dWAxiR"
/>
`;
